### ISSUE TITLE
EXP - Review

### ISSUE THREAD

**Issue by reecepbcups** [notional-labs/craft#133]
**Need guides on flow of commands, was not able to mint EXP in my test environment (go command cli panics, chain still operational)**

### proposal
when making the proposal, user needs to specify a vesting period for the EXP. 
They can stake during this time, but can not burn it to the other asset until that time passes

## craft tx exp burnexp [dao_member_address]
https://github.com/notional-labs/craft/blob/bc8c7c614e90d248f5ba291fb7f9921a95768eff/x/exp/types/msgs.go#L79
- This message burns the entire users EXP, when it should allow for a user to burn partial of their EXP.

Like so:
craft tx exp burnexp [dao_member_address] [exp_to_burn_amount]

## What is the difference between "tx exp spend [coins]" & "tx exp fund [coins]" as 2 separate commands?
- It seems spend [coins] should be the only command, and should do the "fund [coins]" logic all in 1 right?

## craftd tx exp adjust [price]
- So pricing is required to be done manually? What about a smart contract which queries the balances of DAO wallets, and returns the value in USD (coingeko, LP pools in osmo or something). Not sure how Luna did it, not my specialty yet

## craft tx exp send [amount]
"send [amount] from module escrow to the DAO address" so the DAO has some EXP in a module account it uses for sending?

**Comment by vuong177** [comment]
@Reecepbcups adjust is a temporary solution. I'm writing an oracle logic for query exp price.

**Comment by vuong177** [comment]
>proposal
when making the proposal, user needs to specify a vesting period for the EXP.
They can stake during this time, but can not burn it to the other asset until that time passes<

I think `vesting period for the EXP` is a chain-param for every user.

**Comment by Reecepbcups** [comment]
@vuong177 

> Token when a user spends to join DAO will go to exp module escrow instead of the DAO address. `craft tx exp send [amount]` for sending token from exp module escrow to DAO address.
>
<p>So `tx exp spend [coins]` is really tx exp exchange [coins]` where coins = some IBC asset, and in return they get EXP.
Then the DAO will `craft tx exp send [amount]` to send those IBC tokens to a DAO controlled multisig wallet for example to stake / LP / do what ever with correct?</p>
---

> I think `vesting period for the EXP` is a chain-param for every user. Every DAO member have same `vesting period for the EXP`

Is that able to be specified in the proposal? where we mint the tokens to the user as vested for X amount of time? (the proposal can do it at a blockheight or ISO8061, whichever is easier)
or are you saying ALL have the same vesting period time with no way to change that param?

**Comment by vuong177** [comment]
This param could be change by a gov.

**Comment by vuong177** [comment]
Like unbonding time in staking.

**Comment by vuong177** [comment]
> So `tx exp spend [coins]` is really `tx exp exchange [coins]` where coins = some IBC asset, and in return they get EXP. Then the DAO will `craft tx exp send [amount]` to send those IBC tokens to a DAO controlled multisig wallet for example to stake / LP / do what ever with correct?

Yes, this is logic in exp module. I think need to change cli for better readable.

**Comment by vuong177** [comment]
@Reecepbcups I think oracle exp has been done. I create an ibc-channel between craft-testnet <-> bandchain and it working good. If you want to test more, I can provide an RPC for this testnet.
